adminpages/reports/history.php

```diff
@@ -0,0 +1,248 @@
+				<option value="" <?php if( ! $l ) { ?>selected="selected"<?php } ?>><?php _e( 'All Levels', 'paid-memberships-pro'); ?></option>
+				<?php
+					$levels = $wpdb->get_results("SELECT id, name FROM $wpdb->pmpro_membership_levels ORDER BY name");
+					foreach( $levels as $level ) { ?>
+						<option value="<?php echo $level->id; ?>" <?php if ( $l == $level->id ) { ?>selected="selected"<?php } ?>><?php echo esc_html( $level->name ); ?></option>
+					<?php } ?>
+			</select>
+		</li>
+	</ul>
+	<p class="search-box">
+		<label class="hidden" for="post-search-input"><?php _e( 'Search Members', 'paid-memberships-pro' );?>:</label>
+		<input type="hidden" name="page" value="pmpro-reports" />
+		<input type="hidden" name="report" value="member_value_history" />
+		<input id="post-search-input" type="text" value="<?php echo esc_attr( $s ); ?>" name="s" />
+		<input class="button" type="submit" value="<?php esc_html_e( 'Search Members', 'paid-memberships-pro' ); ?>" />
+	</p>
+	<?php
+		//some vars for the search
+		if ( isset( $_REQUEST['pn'] ) ) {
+			$pn = intval( $_REQUEST['pn'] );
+		} else {
+			$pn = 1;
+		}
+
+		if ( isset( $_REQUEST['limit'] ) ) {
+			$limit = intval( $_REQUEST['limit'] );
+		} else {
+			/**
+			 * Filter to set the default number of items to show per page
+			 * on the Members List page in the admin.
+			 *
+			 * @since 1.8.4.5
+			 *
+			 * @param int $limit The number of items to show per page.
+			 */
+			$limit = apply_filters( 'pmpro_memberslist_per_page', 15 );
+		}
+
+		$end = $pn * $limit;
+		$start = $end - $limit;
+
+		$sqlQuery = "SELECT SQL_CALC_FOUND_ROWS u.ID, u.user_login, u.user_email, UNIX_TIMESTAMP(u.user_registered) as joindate, mu.membership_id, UNIX_TIMESTAMP(mu.startdate) as startdate, UNIX_TIMESTAMP(mu.enddate) as enddate, m.name as membership, SUM(mo.total) as totalvalue FROM $wpdb->users u LEFT JOIN $wpdb->pmpro_memberships_users mu ON u.ID = mu.user_id LEFT JOIN $wpdb->pmpro_membership_levels m ON mu.membership_id = m.id LEFT JOIN $wpdb->pmpro_membership_orders mo ON u.ID = mo.user_id LEFT JOIN $wpdb->usermeta um ON u.ID = um.user_id";
+
+		if ( $s ) {
+			$sqlQuery .= " WHERE mu.membership_id > 0 AND (u.user_login LIKE '%" . esc_sql( $s ) . "%' OR u.user_email LIKE '%" . esc_sql( $s ) . "%' OR um.meta_value LIKE '%" . esc_sql( $s ) . "%') ";
+		} else {
+			$sqlQuery .= " WHERE mu.membership_id > 0  ";
+		}
+		
+
+		if( $l ) {
+			$sqlQuery .= " AND mu.status = 'active' AND mu.membership_id = '" . esc_sql( $l ) . "' ";
+		} else {
+			$sqlQuery .= " AND mu.status = 'active' ";
+		}
+		
+		$sqlQuery .= " AND mo.gateway_environment = '" . pmpro_getOption( 'gateway_environment' ) . "' ";
+		$sqlQuery .= " AND mo.status NOT IN('token','review','pending','error','refunded') ";
+		
+		$sqlQuery .= "GROUP BY u.ID ORDER BY totalvalue DESC ";
+
+		$sqlQuery .= "LIMIT $start, $limit";
+
+		$sqlQuery = apply_filters( 'pmpro_members_list_sql', $sqlQuery );
+
+		$theusers = $wpdb->get_results( $sqlQuery );
```

Comment:
```suggestion
		$sqlQuery = "
			SELECT
				SQL_CALC_FOUND_ROWS
				u.ID,
				u.user_login,
				u.user_email,
				UNIX_TIMESTAMP(u.user_registered) AS joindate,
				mu.membership_id,
				UNIX_TIMESTAMP(mu.startdate) AS startdate,
				UNIX_TIMESTAMP(mu.enddate) AS enddate,
				m.name AS membership,
				SUM(mo.total) AS totalvalue
			FROM $wpdb->users AS u
			LEFT JOIN $wpdb->pmpro_memberships_users AS mu
				ON u.ID = mu.user_id
			LEFT JOIN $wpdb->pmpro_membership_levels AS m
				ON mu.membership_id = m.id
			LEFT JOIN $wpdb->pmpro_membership_orders AS mo
				ON u.ID = mo.user_id
			LEFT JOIN $wpdb->usermeta AS um
				ON u.ID = um.user_id
			WHERE
				mu.membership_id > 0
				AND mu.status = 'active'
				AND mo.status NOT IN ( 'token', 'review', 'pending', 'error', 'refunded' )
				AND mo.gateway_environment = %s
		";

		$to_prepare = [
			pmpro_getOption( 'gateway_environment' ),
		];

		if ( $s ) {
			$sqlQuery .= '
				AND (
					u.user_login LIKE %s
					OR u.user_email LIKE %s
					OR um.meta_value LIKE %s
				)
			';

			// We use $wpdb->esc_like() to escape the string to prevent additional % usage.
			$to_prepare[] = '%' . $wpdb->esc_like( $s ) . '%';
			$to_prepare[] = '%' . $wpdb->esc_like( $s ) . '%';
			$to_prepare[] = '%' . $wpdb->esc_like( $s ) . '%';
		}

		if ( $l ) {
			$sqlQuery .= '
				AND mu.membership_id = %s
			';

			$to_prepare[] = $l;
		}

		$sqlQuery .= '
			AND mu.membership_id = %s
		';

		$to_prepare[] = $l;

		$sqlQuery .= '
			GROUP BY u.ID ORDER BY totalvalue DESC
			LIMIT %d, %d
		';

		$to_prepare[] = $start;
		$to_prepare[] = $limit;

		/**
		 * Allow filtering the Members List SQL query.
		 *
		 * @since TBD
		 *
		 * @param string $sqlQuery The Members List SQL query.
		 */
		$sqlQuery = apply_filters( 'pmpro_members_list_sql', $sqlQuery );

		$theusers = $wpdb->get_results( $sqlQuery );
```

Reply:
This whole file was removed recently. Sorry for the wasted work here. We need to fetch the 19 upstream commits from the dev branch.